### ISSUE TITLE
Implement buffering for downloading / playing songs

### ISSUE THREAD

**Issue by Skarkii** [Skarkii/Discord-MusicBot#2]
Currently the bot needs to download the entire song before starting to play, this can cause long delays if an larger file is requested.

Another issue that is caused of this is that on slower CPU's queuing a song may cause stuttering in the active song.

**Comment by Skarkii** [comment]
Implemented in d1766106805045f4864d7f42056e368f3da81e41.

**Comment by Skarkii** [comment]
Closed.
